Build both scap versions, but 1.3 is default

yuumasato opened the pull request:
#### Description:

- Follow up from #4689, build both SCAP versions, instead of only one of them.
- Default means that the datastream named `ssg-<product>-ds.xml` is SCAP 1.3.
- The SCAP 1.2 datastream will be named `ssg-<product>-ds-1.2.xml`.
- Updated `build_script` command with option to generate fully compliant SCAP 1.2 with `--oval510`
- Docs also updated

EDIT:
- Also, `nist-zipfile` target changed to `vendor-zipfile` and documented.

#### Rationale:
- SCAP 1.3 is still not widely supported, until then build SCAP 1.2 content

ggbecker commented:
Please add ctest for datastream validation to 1.2 datastream:

in SSGCommon.cmake:
```

    add_test(
        NAME "validate-ssg-${PRODUCT}-ds-1.2.xml"
        COMMAND "${OPENSCAP_OSCAP_EXECUTABLE}" ds sds-validate "${CMAKE_BINARY_DIR}/ssg-${PRODUCT}-ds-1.2.xml"
    )
```

yuumasato commented:
> Please add ctest for datastream validation to 1.2 datastream:

Should be addressed in b7e4437 

I also updated the cmake commands for the derivative products.

ggbecker commented on cmake/SSGCommon.cmake:
```diff
@@ -571,7 +577,7 @@ macro(ssg_build_sds PRODUCT)
             DEPENDS "${CMAKE_BINARY_DIR}/ssg-${PRODUCT}-cpe-dictionary.xml"
             DEPENDS "${CMAKE_BINARY_DIR}/ssg-${PRODUCT}-cpe-oval.xml"
-            COMMENT "[${PRODUCT}-content] generating ssg-${PRODUCT}-ds.xml"
+            COMMENT "[${PRODUCT}-content] generating ssg-${PRODUCT}-ds.xm and lssg-${PRODUCT}-ds-1.2.xml"
```
```suggestion
            COMMENT "[${PRODUCT}-content] generating ssg-${PRODUCT}-ds.xm and ssg-${PRODUCT}-ds-1.2.xml"
```

ggbecker commented:
As the datastream naming convention is changing, please change here as well:

https://github.com/ComplianceAsCode/content/blob/365a295ca049429e3d7611296b03f94e9b671a6e/tests/run_scapval.py#L112

yuumasato commented:
@ggbecker Nice catch, should be fixed.

ggbecker commented on cmake/SSGCommon.cmake:
```diff
@@ -571,7 +577,7 @@ macro(ssg_build_sds PRODUCT)
             DEPENDS "${CMAKE_BINARY_DIR}/ssg-${PRODUCT}-cpe-dictionary.xml"
             DEPENDS "${CMAKE_BINARY_DIR}/ssg-${PRODUCT}-cpe-oval.xml"
-            COMMENT "[${PRODUCT}-content] generating ssg-${PRODUCT}-ds.xml"
+            COMMENT "[${PRODUCT}-content] generating ssg-${PRODUCT}-ds.xm and ssg-${PRODUCT}-ds-1.2.xml"
```
```suggestion
            COMMENT "[${PRODUCT}-content] generating ssg-${PRODUCT}-ds.xml and ssg-${PRODUCT}-ds-1.2.xml"
```

ggbecker commented on CMakeLists.txt:
```diff
@@ -367,11 +354,11 @@ endif()
     ssg_build_nist_zipfile("scap-security-guide-${SSG_VERSION}-SCAP-1.3")
 else()
-    ssg_build_zipfile("scap-security-guide-${SSG_VERSION}-SCAP-1.2")
+    ssg_build_zipfile("scap-security-guide-${SSG_VERSION}-oval-${SSG_TARGET_OVAL_VERSION}")
```
`${SSG_TARGET_OVAL_VERSION}` contains `dot` in the value (`5.11`). Release artifacts for OVAL 5.10 are in the form of `scap-security-guide-0.1.45-oval-510.zip`. I wonder if this can impact anyone somehow.

ggbecker replied:
```suggestion
    ssg_build_zipfile("scap-security-guide-${SSG_VERSION}-oval-${SSG_TARGET_OVAL_MAJOR_VERSION}${SSG_TARGET_OVAL_MINOR_VERSION}")
```
Would this be acceptable?

yuumasato replied:
I just reverted to state before #4689 
https://github.com/ComplianceAsCode/content/pull/4689/files#diff-af3b638bc2a3e6c650974192a53c7291L268

This is a nice catch, as up to version 0.1.40, the 5.10 zip was named suffix `oval-5.10.zip`.

Do you feel strongly about this change?

ggbecker replied:
It seems that no harm was caused by using a different names. So I'm ok leaving as it is right now.

ggbecker commented on cmake/SSGCommon.cmake:
```diff
@@ -1195,34 +1211,68 @@ macro(ssg_build_zipfile_target ZIPNAME)
+    # When SCAP1.2 zipfiles are needed no more, these custom commands can be unified
+    if("${ZIPNAME}" MATCHES "SCAP-1.3")
+        # Zip only the SCAP1.3 datastream
+        add_custom_command(
```
The only different thing from these two custom commands is the suffix `ds-1.2.xml`. Can you infer from `ZIPNAME` the right suffix and deduplicate the code using some kind of CMake variable?

yuumasato replied:
Good idea, I'll look into it.

yuumasato replied:
It works, thanks for idea.

ggbecker commented:
It's also a good opportunity to mention the target `nist-zipfile`, please add a section next to regarding the target availability:
https://github.com/ComplianceAsCode/content/blob/master/docs/manual/developer_guide.adoc#optional-building-a-zip-file

yuumasato commented:
> It's also a good opportunity to mention the target `nist-zipfile`, please add a section regarding the target availability next to:
> https://github.com/ComplianceAsCode/content/blob/master/docs/manual/developer_guide.adoc#optional-building-a-zip-file

@ggbecker This is a bit of scope creep, but I'm taking it.

I have changed target name `nist-zipfile` to `vendor-zipfile`, and added a target specific for Red Hat products.

ggbecker commented on cmake/SSGCommon.cmake:
```diff
@@ -1251,6 +1251,6 @@ macro(ssg_build_vendor_zipfile ZIPNAME)
         DEPENDS "${CMAKE_BINARY_DIR}/vendor-zipfile/${ZIPNAME}-RedHat.zip"
     )
-    add_custom_target(vendor-zipfile ALL)
+    add_custom_target(vendor-zipfile)
```
I was wondering why my build was always generating the vendor-zipfile when triggering ALL build. ;P

yuumasato commented:
@ggbecker Thank you for the great review!
